documentation/status-proxy-service.md

```diff
@@ -0,0 +1,123 @@
+# status-proxy-service
+
+## Use Case
+The `status-proxy-service` acts as a bridge between HTTP clients and an HTTPS
+`status-service`. It is configured with SSL information about a 
+`status-service`, and then accepts HTTP requests and proxies them over HTTPS 
+to the real `status-service`. 
+
+This could be useful for instance, if you have a load balancer that needs to
+access the `status-service`, but which cannot be configured to use custom SSL
+certs from pem files.
+
+## Preferred Alternatives
+Given that the `status-proxy-service` opens a small hole in your security,
+consider the following alternatives first:
+
+* Configure your load balancer to use HTTPS with pem files if possible
+* If you are comfortable setting up reverse proxying with a tool such as nginx,
+  consider using that
+* Consider running the regular status service on a plaintext port, so that only
+  the status service is accessible via plaintext, and you are not reverse
+  proxying any requests to the HTTPS portions of the apps
+
+## Security Considerations
+There is some security risk associated with providing a plaintext window into
+a portion of your HTTPS endpoints.
+  
+* If incorrectly configured, the `status-proxy-service` could unintentionally
+  open up endpoints other than the `status-service` to plaintext access
+* The proxy should be run on a network interface that is not accessible from
+  outside your internal LAN if possible
+* The proxy should be run on a port that is firewalled to only allow access
+  from your load balancer
+
+## Examples
+
+### Configuring the status-service for plaintext access
+
+If you would like to configure your status service to be on a plaintext port,
+your trapperkeeper configuration might look something like this:
+
+```
+webserver: {
+  default: {
+    ssl-port: 9001
+    ssl-cert: ./ssl/certs/localhost.pem
+    ssl-key: ./ssl/private_keys/localhost.pem
+    ssl-ca-cert: ./ssl/certs/ca.pem
+    default-server: true
+  }
+
+  status: {
+    port: 8080
+  }
+}
+
+web-router-service: {
+    "puppetlabs.trapperkeeper.services.status.status-service/status-service": {
```

Comment:
Cosmetic - would be good for the alignment in the `web-router-service` settings to be consistent with those in the `webserver` section - either 2 or 4 leading spaces.